tests/unit/src/test/scala/tests/markdown/LinkHygieneSuite.scala

```diff
@@ -101,40 +102,81 @@ class LinkHygieneSuite extends FunSuite {
       |/a.md
       |![i](/static/logo.png)
     """.stripMargin,
-    ""
+    List.empty
   )
 
+  val testContent2 = "[absolute](/absolute.md)"
   check(
     "absolute",
-    """
-      |/a.md
-      |[absolute](/absolute.md)
-    """.stripMargin,
-    """|warning: a.md:1:1: Unknown link '/absolute.md', did you mean 'a.md'? To fix this problem, either make the link relative or turn it into complete URL such as http://example.com/absolute.md.
-       |[absolute](/absolute.md)
-       |^^^^^^^^^^^^^^^^^^^^^^^^
-       |""".stripMargin
+    s"""
+       |/a.md
+       |$testContent2""".stripMargin,
+    List(
+      DeadLinkInfo(
+        Position.Range(Input.VirtualFile("a.md", testContent2), 0, 24),
+        "Unknown link '/absolute.md', did you mean 'a.md'? To fix this problem, either make the link relative or turn it into complete URL such as http://example.com/absolute.md."
+      )
+    )
   )
 
+  val testContent3 =
+    """|# Header 1
+       |[2](b.md#header)
+       |""".stripMargin
   check(
     "verbose",
-    """
-      |/a.md
-      |# Header 1
-      |[2](b.md#header)
-      |/b.md
-      |# Header 2
+    s"""
+       |/a.md
+       |$testContent3
+       |/b.md
+       |# Header 2
     """.stripMargin,
-    """|warning: a.md:2:1: Unknown link 'b.md#header', did you mean 'b.md#header-2'?
-       |isValidHeading:
-       |  a.md
-       |  a.md#header-1
-       |  b.md
-       |  b.md#header-2
-       |[2](b.md#header)
-       |^^^^^^^^^^^^^^^^
-       |""".stripMargin,
+    List(
+      DeadLinkInfo(
+        Position.Range(Input.VirtualFile("a.md", testContent3), 11, 27),
+        """|Unknown link 'b.md#header', did you mean 'b.md#header-2'?
+           |isValidHeading:
+           |  a.md
+           |  a.md#header-1
+           |  b.md
+           |  b.md#header-2""".stripMargin
+      )
+    ),
     verbose = true
   )
 
+  test("console rendering") {
+    val myOut = new ByteArrayOutputStream()
+    val reporter = new ConsoleReporter(new PrintStream(myOut))
+    val original =
+      s"""|/a.md
+          |$testContent1""".stripMargin
+    myOut.reset()
+    reporter.reset()
+    val root = tests.cli.StringFS.fromString(original)
+    val settings = Settings
+      .default(root)
+      .copy(reportRelativePaths = true, in = List(root), out = List(root))
+    val links = DocumentLinks.fromGeneratedSite(settings)
+    val result = LinkHygiene.lint(links, false)
+    val resultVerbose = LinkHygiene.lint(links, true)
+    LinkHygiene.report(asError = true, result, reporter)
+    LinkHygiene.report(asError = false, result, reporter)
+    val obtained = fansi.Str(myOut.toString()).plainText
+    val expected =
+      """|error: a.md:3:7: Unknown link 'a.md#does-not-exist'.
```

Comment:
Will this test be needed if we revert the other tests to use string diffs?

Reply:
no, and you're right, when I implemented this test, I was like: wow I could have left the test untouched for the most part and just wrap with `LinkHygiene.report(...)`